proposals/hasCrossSiteAncestor.md

```diff
@@ -0,0 +1,113 @@
+# Proposal: Update Extensions to interact with Partitioned Cookies containing Cross-Site Ancestor Chain Bit values
+
+**Summary**
+
+Allow extensions to utilize a hasCrossSiteAncestor boolean value when interacting with partitioned cookies that include a cross-site ancestor chain bit in their partiton key.
+
+#### Brackground information: Description of Cross-site Ancestor Chain Bit in partitioned cookies
+
+A cross-site ancestor chain bit is a value that is set by the browser when a paritioned cookie is created. The value is used as part of the key in the partition key to determine access to a partitioned cookie. If the bit indicates true, if the cookie has been set in a third third party context.  
+
+A third party context occurs when any parent frame, in the frame tree, is cross-site with the frame the cookie is being set on. If one site of the frame tree is third party, then all children frames of that frame are also third party and have a cross site ancestor.
+
+
+| Site frame tree |Site cookie is set on| hasCrossSiteAncestor value of cookie|
+|---|---|--|
+| A1         |A1| false         | 
+| A1->A2      |A2 | false          | 
+| A1->A2->A3   |A3   | false          | 
+| A1->B       |B| true          |
+| A1->B->A2    |A1   | false          |
+| A1->B->A2 |B    | true          |
+| A1->B->A2 |A2   | true          |
+
+**Document Metadata**
+
+**Author:**  aselya 
+
+**Sponsoring Browser:** Chrome
+
+**Contributors:** DCtheTall
+
+**Created:** 2024-04-01
+
+**Related Issues:** [PrivacyCG/CHIPS issue 6 - How do Partitioned cookies interact with browser extensions?](https://github.com/privacycg/CHIPS/issues/6)
+
+## Motivation
+https://github.com/privacycg/CHIPS/issues/40 is adding a cross-site ancestor bit value to partitioned cookies.
+
+### Objective
+To interact with partitioned cookies containing a cross-site ancestor chain bit correctly, extensions will need to have the ability to to specify a value (hasCrossSiteAncestor) that corresponds to the value of the cross-site ancestor chain bit in partitioned cookies.
+
+#### Use Cases
+
+#### Password Manager:
+Let’s say a password manager extension (with host permissions) is used by users to access their login information by setting a cookie that stores their usernames and passwords in an encrypted partitioned cookie. To protect their users against clickjacking, the extension adds a setting that prevents their cookies from being accessed, by default, in embeds that have cross site ancestors without triggering a user prompt. If permission is given through the prompt, the extension sets a cookie with a hasCrossSiteAncestor value of true. Upon subsequent visits, the extension checks the cookie store for the presence of a cookie with a hasCrossSiteAncestor value of true to determine whether the prompt needs to be rendered.
```

Comment:
Hmm, I didn't find this example particularly convincing:

- Is the extension storing the cookies on a chrome-extension:// page? If so, that wouldn't work because chrome-extension:// schemes aren't considered "secure" for cookie purposes and therefore you aren't able to set the `Partitioned` attribute.
- Alternatively, is the suggestion a password manager would use encrypted cookies on the origin you are signing in to? I'm not sure that's a realistic path we would encourage developers down.

I wonder if we could go with something simpler like a cookie manager extension that wants to give users full insight into all of the cookies that might be stored.

Reply:
I think that in this use case, the "password manager" extension uses cookies associated with a third party website as a storage mechanism. Odd use case but I find it useful to illustrate why this could be desirable for an extension API to have this functionality.

Reply:
I don't think there's any harm in including two use cases. I'll add @oliverdunk's example to the PR.